graphql-java-servlet/src/main/java/graphql/servlet/AbstractGraphQLHttpServlet.java

```diff
@@ -50,31 +50,66 @@
     private static final GraphQLRequest INTROSPECTION_REQUEST = new GraphQLRequest(IntrospectionQuery.INTROSPECTION_QUERY, new HashMap<>(), null);
     private static final String[] MULTIPART_KEYS = new String[]{"operations", "graphql", "query"};
 
+    private GraphQLConfiguration configuration;
+
+    /**
+     * @deprecated override {@link #getConfiguration()} instead
+     */
+    @Deprecated
     protected abstract GraphQLQueryInvoker getQueryInvoker();
 
+    /**
+     * @deprecated override {@link #getConfiguration()} instead
+     */
+    @Deprecated
     protected abstract GraphQLInvocationInputFactory getInvocationInputFactory();
 
+    /**
+     * @deprecated override {@link #getConfiguration()} instead
+     */
+    @Deprecated
     protected abstract GraphQLObjectMapper getGraphQLObjectMapper();
 
-    private final List<GraphQLServletListener> listeners;
+    /**
+     * @deprecated override {@link #getConfiguration()} instead
+     */
+    @Deprecated
+    protected abstract boolean isAsyncServletMode();
+
+    protected GraphQLConfiguration getConfiguration() {
+        return GraphQLConfiguration.with(getInvocationInputFactory())
+                .with(getQueryInvoker())
+                .with(getGraphQLObjectMapper())
+                .with(isAsyncServletMode())
+                .with(listeners)
+                .build();
+    }
 
-    private final HttpRequestHandler getHandler;
-    private final HttpRequestHandler postHandler;
+    /**
+     * @deprecated use {@link #getConfiguration()} instead
+     */
+    @Deprecated
+    private final List<GraphQLServletListener> listeners;
 
-    private final boolean asyncServletMode;
+    private HttpRequestHandler getHandler;
+    private HttpRequestHandler postHandler;
 
     public AbstractGraphQLHttpServlet() {
-        this(null, false);
+        this(null);
     }
 
-    public AbstractGraphQLHttpServlet(List<GraphQLServletListener> listeners, boolean asyncServletMode) {
+    public AbstractGraphQLHttpServlet(List<GraphQLServletListener> listeners) {
         this.listeners = listeners != null ? new ArrayList<>(listeners) : new ArrayList<>();
-        this.asyncServletMode = asyncServletMode;
+    }
+
+    @Override
+    public void init(ServletConfig servletConfig) {
+        this.configuration = getConfiguration();
 
         this.getHandler = (request, response) -> {
```

Comment:
These handlers should be extracted and no longer rely on `HttpServletRequest` or `HttpServletResponse`. They should function independently. That way they can probably be moved down into the `common-http` module.